### ISSUE TITLE
[BUG] Incognito Browser directly logs in to SSO enabled Internal Webpage skipping the Login

### ISSUE THREAD

**Issue by sureshahead** [microsoft/playwright#9068]
**Context:**
Playwright Version: 1.15.0
Operating System: Windows 10 Enterprise (64-bit)
Browser: Chrome & Edge (having issue loading Firefox #8957)

**Describe the bug**

We need to automate the testing of Internal Website which is SSO enabled. We see that PW starts the browser in incognito (which is our req. aswell) but the page loads using the current SSO user cred thus skipping the Login screen. One weird thing is that, three of us are exploring PW - out of which couple of us are on Windows enterprise and one is on Mac, the guy on Mac is getting SSO login screen for the same URL while running the PW Script. 

**Other useful info :** we tried to debug the script and while in debug, we tried to open Gmail in the browser (in the instance opened by PW) and in that case we see Gmail Login even though am logged into Gmail in my chrome. Also from the PW opened Chrome browser we opened a new incognito manually and tried to load our Internal App URL which shows the SSO Login screen. Not sure if the PW opened incognito is any difference in preference.

Please try this scenario in any of your MSFT internal app which will open Login screen when accessed through Incognito.

We are evaluating PW for our automation purpose and this is a blocker to get started as we have test users with different access roles which cannot be tested if we don't get the SSO Login screen. Happy to help you with any specific logs incase needed. Looking forward.

**Comment by mxschmitt** [comment]
Could you clarify "SSO" a bit more? Is it about client certificates (https://github.com/microsoft/playwright/issues/1799) or is it e.g. about SAML/oAuth which gets declined because the authentification provider does not like our incognito tabs?

**Comment by sureshahead** [comment]
To put it functionally : its the other way. 
There is internal web app for which I have access to with my Single sign-on. What we do during automation testing [using selenium] is to open the browser in incognito, when we access the internal URL - we are loaded with Login screen where we use Test User Accounts to logon based on the use case. Whereas incase of PlayWright - we see that the incognito browser opening the internal webapp with my user credential already logged in which is not intended [we tested with edge aswell and same behavior]

As I mentioned in the initial bug description we see that this is happening in Windows Chrome and Edge whereas in Mac chrome browser the page is loaded with login page as intended. 

Hope this helps!!

**Comment by mxschmitt** [comment]
So to summarize, this is only reproducible for Chrome/Edge on Windows and the issue is about that you are already logged in on that environments for some reason. So the state is kinda persistent. How do you consume Playwright. Are you using the Library or via the test-runner?

I guess there are Enterprise rules applied which lead to that behavior. Could you navigate to `chrome://policy` in Chrome and see if there are some mentioned?

We would need to reproduce it or find out which rule is causing that to find out the reason.

Thanks!

**Comment by sureshahead** [comment]
@mxschmitt : am attaching the screenshot of policy for your reference here. One thing to note as I mentioned earlier - from the PW opened Incognito Chrome browser, we opened a new incognito manually and tried to load our Internal App URL which shows the SSO Login screen. Not sure if the PW opened incognito is any difference in preference.

![enterpride_policy](https://user-images.githubusercontent.com/57895333/135320946-4c63baa6-6ca1-470e-91a7-6eaaac71d342.png)


**Playwright opened incognito looks with white BG startup page (not sure if this is of significance but just pointing here) :**
![playwright_opened_incognito](https://user-images.githubusercontent.com/57895333/135321001-3ed2b8bf-17fe-45ea-981c-b95373abaa74.png)

**From the PW opened incognito we opened the Incognito manually and it loads with black BG startup page (also the SSO Login page is displayed when we hit the internal APP URL) :**
![manually_opened_incognito](https://user-images.githubusercontent.com/57895333/135321208-f40441ab-8719-458e-b9c7-4d2be5434102.png)

**Comment by JoelEinbinder** [comment]
@pavelfeldman do you know what global setting we could be reading that would cause this? I know our incognito is not really incognito sometimes, but I am not sure about the specifics.

**Comment by badyornyi** [comment]
Having same issue on corporate VM under Win10.

I found workaround, @sureshahead , probably it may help you:
1. Goto "Internet Options"
2. Tab "Security"
3. [For both "Internet" and "Local intranet"] click "Custom level" (for "Trusted sites" I have not permissions on my corp VM to change)
4. Scroll "Settings" list to the bottom
5. For "User Authentication > Logon" set "Prompt for user name and password"
6. After this changes I can successfully use different users to log in in my app via Chrome/Edge

ps: I don't have this issue on my Mac too.

**Comment by sureshahead** [comment]
Much appreciated @badyornyi. I will certainly give this a try. Btw we will run into problem when the tests need to be run in saucelabs/browserstack. Just hoping that this gets fixed sometime sooner with some reproducible steps for the PW team. Until then the local fix certainly helps to unblock the dev's using windows.

**Comment by zenntenn** [comment]
We're having this issue with a public facing website. We're using Windows to run Playwright, and it's Playwright .Net.

Website is https://as-esdccms-testing.azurewebsites.net/ , test account is SVAZ-EWCMS-SYSADMIN@hrsdc-rhdcc.gc.ca

If I go with the Chromium version that Playwright installs locally, and I open that manually in an incognito window, it doesn't leak through. If I do the same action in the Playwright window, it does try to use my logged in Windows account.

``` CSharp
        private const string startPage = "https://as-esdccms-testing.azurewebsites.net/";

        [Test]
        public async Task Login()
        {
            // URL is https://as-esdccms-testing.azurewebsites.net/
            var config = new ConfigurationBuilder()
                .AddJsonFile("creds.json", optional: true)
                .AddEnvironmentVariables()
                .Build();

            var username = config["serviceUser"];
            var password = config["password"];

            var playwright = await Playwright.CreateAsync();
            var browser = await playwright.Chromium.LaunchAsync(new BrowserTypeLaunchOptions() { Args = new[] { "--incognito" }, Headless = false });
            var browserContext = await browser.NewContextAsync(new BrowserNewContextOptions() { IgnoreHTTPSErrors = true, HttpCredentials = new HttpCredentials { Username = username, Password = password } });
            var page = await browserContext.NewPageAsync();
            await page.GotoAsync(startPage, new PageGotoOptions { WaitUntil = WaitUntilState.DOMContentLoaded });
            await page.WaitForNavigationAsync();

            await page.FillAsync("input[type='email'][name='loginfmt']", username);

            await page.ClickAsync("input[type='submit']");

            await page.WaitForURLAsync(startPage);
```

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

We are prioritizing the bugs based on the upvotes, recency and our ability to act. It looks like this issue only has a handful of upvotes, has not been touched recently and/or we lack sufficient feedback to act on it. We are closing issues like this one to keep our bug database maintainable. Please feel free to open a new issue and link this one to it if you think this is a mistake.

**Comment by Ramesh1103** [comment]
HI @sureshahead are you able to resolve the issue still I am facing same issue.
i am using playwright 1.30.0 version

**Comment by LakshmiYaswanth** [comment]
Hi @pavelfeldman and @mxschmitt  iam also facing the same issue. Is there any workaround to make this senario work.
Now iam using this playwright version "@playwright/test": "^1.25.2" .

**Comment by ystvan** [comment]
> Having same issue on corporate VM under Win10.
> 
> I found workaround, @sureshahead , probably it may help you:
> 
> 1. Goto "Internet Options"
> 2. Tab "Security"
> 3. [For both "Internet" and "Local intranet"] click "Custom level" (for "Trusted sites" I don't have permissions on my corp VM to change)
> 4. Scroll "Settings" list to the bottom
> 5. For "User Authentication > Logon" set "Prompt for user name and password"
> 6. After this changes I can successfully use different users to log in in my app via Chrome/Edge
> 
> ps: I don't have this issue on my Mac too.

Works like a charm corporate Win10! Thanks
